openhands/cli/utils.py

```diff
@@ -83,8 +83,10 @@ def extract_model_and_provider(model):
     return {'provider': provider, 'model': model_id, 'separator': separator}
 
 
-def organize_models_and_providers(models):
-    result = {}
+def organize_models_and_providers(
+    models: list[str],
+) -> dict[str, dict[str, str | list[str]]]:
+    result: dict[str, dict[str, str | list[str]]] = {}
```

Comment:
When the typing is this complicated its generally better to create a strongly typed object (using pydantic basemodel for instance)